tests/wandb_run_test.py

```diff
@@ -104,6 +134,7 @@ def test_log_code(test_settings):
         f.write("Not that big")
     art = run.log_code()
     assert sorted(art.manifest.entries.keys()) == ["test.py"]
+    run.finish()
```

Comment:
Same here, the `test_settings` fixture has:

```
if wandb.run is not None:
        wandb.run.finish()
```

I'm sure there was a reason for this, but if we can make it "Just Work" in tests it's going to make future devs lives a lot less error prone.

Reply:
Yea, I was trying to find the reason for flakiness and the standing hypothesis was that maybe for some reason the singleton doesn't get recycled properly or something, which is of course not true since all fixtures are func-scoped.
I later discovered that it's (mainly) because of stuff leaking into `os.environ` and the filesystem through certain fixtures and tests, but decided to keep these calls in as it seemed more explicit and a settings fixture didn't seem like the first place I'd look for action like that. 


Now that you bring this up, how about ditching it from `test_settings` and instead doing something like
```python
@pytest.fixture(autouse=True)
def cleanup():
    yield
    if wandb.run is not None:
        wandb.run.finish()
```
What do you think? @raubitsj?